finesse/hardware/plugins/stepper_motor/st10_controller.py

```diff
@@ -182,10 +172,20 @@ def __init__(self, port: str, baudrate: int = 9600, timeout: float = 5.0) -> Non
         SerialDevice.__init__(self, port, baudrate)
 
         self._reader = _SerialReader(self.serial, timeout)
-        self._reader.async_read_completed.connect(self._send_move_end_message)
+        self._reader.async_read_completed.connect(self._on_initial_move_end)
         self._reader.read_error.connect(self.send_error_message)
         self._reader.start()
 
+        self._init_error_timer = QTimer()
+        """A timer to raise an error if the motor takes too long to move."""
+        self._init_error_timer.setInterval(10000)  # 10 seconds
```

Comment:
Why 10s? Is it a sensible amount of time to wait?

Reply:
It could be made a bit shorter, but if it takes longer than 10s then something's definitely gone wrong. (This is the total time taken for the motor to home itself, which involves moving all the way to the optoswitch so it knows which way it's oriented, then rotating so it's facing downwards, which we take as 0°.)

@dc2917 has made the sensible suggestion to make timeout configurable though, so I'll do that.